malcolm/modules/scanning/parts/scanrunnerpart.py

```diff
@@ -0,0 +1,498 @@
+        registrar.add_attribute_model(
+            "runnerStatusMessage",
+            self.runner_status_message,
+            self.runner_status_message.set_value)
+
+        self.scan_file = StringMeta(
+            "Path to input scan file",
+            tags=[config_tag(), Widget.TEXTINPUT.tag()]
+        ).create_attribute_model()
+        registrar.add_attribute_model(
+            "scanFile",
+            self.scan_file,
+            self.scan_file.set_value)
+
+        self.scans_configured = NumberMeta(
+            "int64", "Number of scans configured",
+            tags=Widget.TEXTUPDATE.tag()
+        ).create_attribute_model()
+        registrar.add_attribute_model(
+            "scansConfigured",
+            self.scans_configured,
+            self.scans_configured.set_value)
+
+        self.current_scan_set = StringMeta(
+            "Current scan set",
+            tags=Widget.TEXTUPDATE.tag()
+        ).create_attribute_model()
+        registrar.add_attribute_model(
+            "currentScanSet",
+            self.current_scan_set,
+            self.current_scan_set.set_value)
+
+        self.scans_completed = NumberMeta(
+            "int64", "Number of scans completed",
+            tags=Widget.TEXTUPDATE.tag()
+        ).create_attribute_model()
+        registrar.add_attribute_model(
+            "scansCompleted",
+            self.scans_completed,
+            self.scans_completed.set_value)
+
+        self.scan_successes = NumberMeta(
+            "int64", "Successful scans",
+            tags=[Widget.TEXTUPDATE.tag()]
+        ).create_attribute_model()
+        registrar.add_attribute_model(
+            "scanSuccesses",
+            self.scan_successes,
+            self.scan_successes.set_value)
+
+        self.scan_failures = NumberMeta(
+            "int64", "Failed scans",
+            tags=[Widget.TEXTUPDATE.tag()]
+        ).create_attribute_model()
+        registrar.add_attribute_model(
+            "scanFailures",
+            self.scan_failures,
+            self.scan_failures.set_value)
+
+        self.output_directory = StringMeta(
+            "Root output directory (will create a sub-directory inside)",
+            tags=[config_tag(), Widget.TEXTINPUT.tag()]
+        ).create_attribute_model()
+        registrar.add_attribute_model(
+            "outputDirectory",
+            self.output_directory,
+            self.output_directory.set_value)
+
+        # Methods
+        registrar.add_method_model(self.loadFile)
+        registrar.add_method_model(self.run, needs_context=True)
+
+    def get_file_contents(self):
+        # type: () -> str
+
+        try:
+            with open(self.scan_file.value, "r") as input_file:
+                return input_file.read()
+        except IOError:
+            self.set_runner_state(RunnerStates.FAULT)
+            self.runner_status_message.set_value("Could not read scan file")
+            raise IOError("Could not read scan file")
```

Comment:
```suggestion
            raise
```
I would just raise here, as the error message from open() is probably more descriptive and the traceback is useful to have

Reply:
I have changed this accordingly.